### ISSUE TITLE
Potential erroneous caching in cypress cloud

### ISSUE THREAD

**Issue by luanesouza** [cypress-io/cypress#26426]
### Current behavior

I have a list of pages that I test, which are not behind auth0. These are **public** routes.

The tests were passing as expected in Cypress cloud, but a couple days ago I created a branch to update the auth0-lock package.
That branch was **not** merged, in fact, I **closed** it a couple days later, however since the creation of that branch the tests are **no longer passing** in Cypress cloud, though they pass **locally**.
The error can be seen below.

Please let me know of any clarifying questions you may have.

**Cypress cloud is behaving as if I merged the auth0-lock PR.**
![image](https://user-images.githubusercontent.com/43556847/229863796-2753ebe4-df20-4a02-a59d-a9a0bb42ea47.png)

### Desired behavior

If the tests are passing locally, I expect them to pass in the cloud as well, since there were no changes introduced to them.
I also expect the auth functionality to not be called, since I am not running any tests in the auth routes.

### Test code to reproduce

The tests only fail in the cloud.
Sample below:
```
cy.visit('http://localhost:3000/squad-intro');
    cy.wait(6000);
    cy.get('[data-test-id=get-started-button]').click({ force: true });
    cy.wait(6000);
```

    cy.get('[data-test-id=roles-container]').should('be.visible');
    cy.get('[data-test-id=checkbox-label]').first().click();
    cy.get('[data-test-id=next-button]').click();
    cy.get('[data-test-id=ideal-start-container]', { timeout: 6000 }).should(
      'be.visible'
    );

### Cypress Version

12.9.0

### Node version

16

### Operating System

Whatever OS cypress clouds uses

### Other

What I tried so far:

- Checked Auth0 logs There are no logs or error messages associated with the failed authentication attempts
- Checked the dependencies update. There have been no changes made to the auth0 package.
- Tested the other branches to see if the auth0 error is branch specific. It is not.
- Added console.logs to auth methods to see if they were being called multiple times. They are not.
- Checked different browsers

**Comment by luanesouza** [comment]
TLDR: The issue seems to be that cypress cloud has cached the PR that updates the auth0-lock package.

**Comment by mabela416** [comment]
Hi @luanesouza I'm taking a look at it

**Comment by luanesouza** [comment]
> Hi @luanesouza I'm taking a look at it

Thanks! Let me know if there's anything else I can provide.

**Comment by mabela416** [comment]
@luanesouza No code is cached on the cypress cloud side. You might want to look into your CI providers caching behavior and see if the package is updated there. It also seems like auth0 has some rate limiting policies, so if you've started to write more tests I'd imagine this error would just recently show up. Consider finding a way to mock authentication without directly auth0 in your tests

**Comment by luanesouza** [comment]
@mabela416 I will take a look at other providers and see if there's anywhere else the problem could live. Thanks for looking into this

**Comment by nagash77** [comment]
Hi @luanesouza , If you haven't read our [Auth0 guide](https://docs.cypress.io/guides/end-to-end-testing/auth0-authentication) I would suggest you check that out as well.  [cy.session](https://docs.cypress.io/api/commands/session) might also be useful depending on your use case.

If you have other questions about best practices or test writing check out our [community chat](https://discord.gg/cypress).

If you narrow down the problem a bit more and it points to Cypress please comment back here and we can reopen this issue and investigate further based on the information.

**Comment by luanesouza** [comment]
> Hi @luanesouza , If you haven't read our [Auth0 guide](https://docs.cypress.io/guides/end-to-end-testing/auth0-authentication) I would suggest you check that out as well. [cy.session](https://docs.cypress.io/api/commands/session) might also be useful depending on your use case.
> 
> If you have other questions about best practices or test writing check out our [community chat](https://discord.gg/cypress).
> 
> If you narrow down the problem a bit more and it points to Cypress please comment back here and we can reopen this issue and investigate further based on the information.

I did read your Auth0 guide, but as I stated in my main post, we are only testing public routes. No auth routes are being tested at this time, hence why I wanted to check if cypress was caching the auth0 package update.
Thanks again for looking into this. I appreciate it.
